internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -64,6 +67,99 @@ func computeCostComponent(region, machineType string, purchaseOption string, ins
+	sustainedUseDiscount := 0.0
+	fixPurchaseOption := "Preemptible"
+	if strings.ToLower(purchaseOption) == "on_demand" {
+		fixPurchaseOption = "OnDemand"
+		switch strings.ToLower(strings.Split(machineType, "-")[0]) {
+		case "c2", "n2", "n2d":
+			sustainedUseDiscount = 0.2
+		case "n1", "f1", "g1", "m1":
+			sustainedUseDiscount = 0.3
+		}
+	}
+
+	var re = regexp.MustCompile(`(\D.+)-(\d+)-(\d.+)`)
+	firstMachineTypeInfo := re.ReplaceAllString(machineType, "$1")
+	strCPUAmount := re.ReplaceAllString(machineType, "$2")
+	strRAMAmount := re.ReplaceAllString(machineType, "$3")
+
+	instanceType := ""
+	if firstMachineTypeInfo != "custom" {
+		instanceType = strings.ToUpper(strings.Split(firstMachineTypeInfo, "-")[0])
+	}
+
+	ext := ""
+	if strings.Contains(strRAMAmount, "ext") {
+		ext = "Extended"
+		strRAMAmount = strings.Split(strRAMAmount, "-")[0]
+	}
+
+	numberOfCores, err := strconv.ParseInt(strCPUAmount, 10, 64)
+	if err != nil {
+		log.Warnf("Could not parse the custom number of Cores for %s", machineType)
+		return nil
+	}
+
+	mbOfRAM, err := strconv.ParseInt(strRAMAmount, 10, 64)
+	if err != nil {
+		log.Warnf("Could not parse the custom amount of Ram for %s", machineType)
+		return nil
+	}
+
+	qty := decimal.NewFromFloat(730)
+	if monthlyHours != nil {
+		qty = decimal.NewFromFloat(*monthlyHours)
+	}
+
+	cpuCostComponent := &schema.CostComponent{
+		Name:                fmt.Sprintf("Custom Instance CPU usage (Linux/UNIX, %s, %s)", purchaseOptionLabel(purchaseOption), machineType),
+		Unit:                "hours",
+		UnitMultiplier:      decimal.NewFromInt(numberOfCores),
+		MonthlyQuantity:     decimalPtr(qty.Mul(decimal.NewFromInt(instanceCount * numberOfCores))),
+		MonthlyDiscountPerc: sustainedUseDiscount,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("gcp"),
+			Region:        strPtr(region),
+			Service:       strPtr("Compute Engine"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "description", ValueRegex: regexPtr(fmt.Sprintf("%s.*Custom Instance Core.*", instanceType))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr(fixPurchaseOption),
+		},
+	}
+
+	ramCostComponent := &schema.CostComponent{
+		Name:                fmt.Sprintf("Custom Instance RAM usage (Linux/UNIX, %s, %s)", purchaseOptionLabel(purchaseOption), machineType),
+		Unit:                "gibibyte hour",
+		UnitMultiplier:      decimal.NewFromInt(mbOfRAM / 1024),
+		MonthlyQuantity:     decimalPtr(qty.Mul(decimal.NewFromInt(instanceCount * mbOfRAM / 1024))),
+		MonthlyDiscountPerc: sustainedUseDiscount,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("gcp"),
+			Region:        strPtr(region),
+			Service:       strPtr("Compute Engine"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "description", ValueRegex: regexPtr(fmt.Sprintf("%s.*Custom.*%s.*Ram.*", instanceType, ext))},
```

Comment:
We should make similar change to this regex as with the one for the Cores above.